distributed/benchmarks/client.py

```diff
@@ -37,3 +33,38 @@ def inc(x):
         total = delayed(sum, pure=True)(L)
 
         total.compute(scheduler=self.client)
+
+
+class WorkerRestrictionsSuite(object):
+
+    def setup(self):
+        cluster = LocalCluster(n_workers=1, threads_per_worker=1,
+                               resources={"resource": 1}, worker_class=Worker)
```

Comment:
For speed reasons we might want to use `processes=False` in cases like this.  It also helps considerably with cleanup.